pykickstart/commands/autopart.py

```diff
@@ -630,3 +630,38 @@ def _getParser(self):
 
                     Partitioning scheme 'btrfs' was removed.""" % versionToLongString(RHEL8)
         return op
+
+class F41_AutoPart(F38_AutoPart):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hw_passphrase = kwargs.get("hw_passphrase", "")
+
+    def __str__(self):
+        retval = super().__str__()
+        if not self.autopart:
+            return retval
+
+        if self.encrypted and self.hw_passphrase:
+            retval = retval.strip()
+            retval += " --hw-passphrase=\"%s\"" % self.hw_passphrase
+            retval += "\n"
+        return retval
+
+    def _getParser(self):
+        op = super()._getParser()
+        op.add_argument("--hw-passphrase", dest="hw_passphrase", version=F41, default="",
+                        help="""
+                        Only relevant if ``--encrypted`` is specified and ``--luks-version``
+                        is set to either `luks2-hw-opal` or `luks2-hw-opal-only`.
+
+                        OPAL administrator passphrase needed to create a new OPAL locking range.""")
+        return op
+
+    def parse(self, args):
+        retval = super().parse(args)
+
+        if self.hw_passphrase and (not self.encrypted or not self.luks_version.startswith("luks2-")):
+            msg = _("OPAL admin passphrase may be specified only with LUKSv2 HW-OPAL encrypted devices")
```

Comment:
Change `OPAL admin passphrase` to `hw-passphrase` so it's clearer what part of the arguments it is talking about.

Reply:
Done.